src/garage/torch/modules/tanh_gaussian_mlp_module.py

```diff
@@ -0,0 +1,242 @@
+        input_dim (int): Input dimension of the model.
+        output_dim (int): Output dimension of the model.
+        hidden_sizes (list[int]): Output dimension of dense layer(s) for
+            the MLP for mean. For example, (32, 32) means the MLP consists
+            of two hidden layers, each with 32 hidden units.
+        hidden_nonlinearity (callable): Activation function for intermediate
+            dense layer(s). It should return a torch.Tensor. Set it to
+            None to maintain a linear activation.
+        hidden_w_init (callable): Initializer function for the weight
+            of intermediate dense layer(s). The function should return a
+            torch.Tensor.
+        hidden_b_init (callable): Initializer function for the bias
+            of intermediate dense layer(s). The function should return a
+            torch.Tensor.
+        output_nonlinearity (callable): Activation function for output dense
+            layer. It should return a torch.Tensor. Set it to None to
+            maintain a linear activation.
+        output_w_init (callable): Initializer function for the weight
+            of output dense layer(s). The function should return a
+            torch.Tensor.
+        output_b_init (callable): Initializer function for the bias
+            of output dense layer(s). The function should return a
+            torch.Tensor.
+        learn_std (bool): Is std trainable.
+        init_std (float): Initial value for std.
+            (plain value - not log or exponentiated).
+        std_hidden_sizes (list[int]): Output dimension of dense layer(s) for
+            the MLP for std. For example, (32, 32) means the MLP consists
+            of two hidden layers, each with 32 hidden units.
+        min_std (float): If not None, the std is at least the value of min_std,
+            to avoid numerical issues (plain value - not log or exponentiated).
+        max_std (float): If not None, the std is at most the value of max_std,
+            to avoid numerical issues (plain value - not log or exponentiated).
+        std_hidden_nonlinearity (callable): Nonlinearity for each hidden layer
+            in the std network.
+        std_hidden_w_init (callable):  Initializer function for the weight
+            of hidden layer (s).
+        std_hidden_b_init (callable): Initializer function for the bias
+            of intermediate dense layer(s).
+        std_output_w_init (callable): Initializer function for the weight
+            of output dense layer(s) in the std network.
+        std_parameterization (str): How the std should be parametrized. There
+            are two options:
+            - exp: the logarithm of the std will be stored, and applied a
+               exponential transformation
+            - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+
+    """
+
+    def __init__(self,
+                 input_dim,
```

Comment:
can't you infer this from the input?

Reply:
Sorry, I'm not sure what you mean by this. This is the constructor for a pytorch nn.module that is the base for the `TanhGaussianPolicy`. Wouldn't one need to construct a policy/module first before passing it some input(such as an observation)

Reply:
i was basically asking why this needs to be a member variable if it's only used in `forward` where it can be inferred from the shape of `input`, but it appears to be a result of the weird inheritance hierarchy of torch.modules more than anything.